libraries/WiFi/src/WiFiMulti.cpp

```diff
@@ -48,10 +48,19 @@ uint8_t WiFiMulti::run(uint32_t connectTimeout)
     int8_t scanResult;
     uint8_t status = WiFi.status();
     if(status == WL_CONNECTED) {
-        for(uint32_t x = 0; x < APlist.size(); x++) {
-            if(!_bStrict || WiFi.SSID()==APlist[x].ssid){
+        if (_bTestConnection && !_bWFMInit){
+            if (testConnection()) {
+                _bWFMInit = true;
                 return status;
             }
+        } else if (!_bStrict) {
+            return status;
+        } else {
+            for(uint32_t x = 0; x < APlist.size(); x++) {
```

Comment:
since you don't need the index value this can be written as:
```
for(auto ap : APlist) {
  if(WiFi.SSID() == ap.ssid) {
    return status;
  }
}
```

Reply:
Much more elegant.  I need to play with vectors more...